activerecord/lib/active_record/database_configurations/database_config.rb

```diff
@@ -10,7 +10,7 @@ class DatabaseConfig # :nodoc:
 
       def self.new(...)
         instance = super
-        instance.adapter_class # Ensure resolution happens early
+        instance.adapter_class if instance.adapter.present? # Ensure resolution happens early
```

Comment:
```suggestion
        instance.adapter_class if instance.adapter # Ensure resolution happens early
```

I don't think we need `present?`.

Reply:
It's not necessary. But I'm trying to be defensive to not give errors that aren't actionable.

```
database configuration specifies nonexistent '' adapter. Ensure that the adapter is spelled correctly in config/database.yml and that you've added the necessary adapter gem to your Gemfile. (ActiveRecord::AdapterNotFound)
```

Empty values in YAML are usually `nil`, but that's not a useful error to receive because some named YAML fragment returns an empty string for `adapter:` because of some quirk of YAML. Empty string will never be valid or useful.

Reply:
These YAML fragments are wrong and should be fixed. Ignoring blank strings in this context makes no sense. Either the key is set or it's not, we shouldn't concern ourselves with the content of the value.

Reply:
Ok! Merged your suggestion.

I'm already working on the fragements.

Reply:
Thanks, please squash the commits thought.